src/cocotb/triggers.py

```diff
@@ -917,13 +906,7 @@ def on_done(ret: Outcome[Any]) -> None:
         for w in waiters:
             w.kill()
 
-        # These lines are the way they are to make tracebacks readable:
-        #  - The comment helps the user understand why they are seeing the
-        #    traceback, even if it is obvious top cocotb maintainers.
-        #  - Using `NullTrigger` here instead of `result = completed[0].get()`
-        #    means we avoid inserting an `outcome.get` frame in the traceback
-        first_trigger = NullTrigger(outcome=completed[0])
-        return await first_trigger
+        return completed[0].get()
```

Comment:
I never really understood this part.
What scenario and traceback was this concerned with?
Is it when the outcome is an `Error` and so `get()` raises?

Reply:
Yep, that's about it. It's just to avoid the `get()` call in the traceback.

Reply:
One thing to note is that in the coming future, `cocotb.start_soon` Tasks will not propagate exceptions, so this is kind of moot.